src/main/scala/atom/Atom.scala

```diff
@@ -0,0 +1,113 @@
+package laughedelic.atom
+
+import scala.scalajs.js, js.annotation._
+
+@js.native
+trait Disposable extends js.Object {
+  def dispose(): Unit = js.native
+}
+
+@js.native
+trait CommandRegistry extends js.Object {
+  def add(target: String, commandName: String, callback: js.UndefOr[js.Any => js.Any]): Unit = js.native
+  def add(target: String, commands: js.Dictionary[js.Any => js.Any], callback: js.UndefOr[js.Any => js.Any]): Unit = js.native
+}
+
+@js.native
+trait Model extends js.Object {
+  def destroy(): Unit = js.native
+  def isDestroyed(): Boolean = js.native
+}
+
+
+@js.native
+trait TextEditor extends Model {
+  val id: Long = js.native
+  
+  val firstVisibleScreenRow: Long = js.native
+  val rowsPerPage: Long = js.native
+
+  // File Details
+  def getTitle(): String = js.native
+  def getLongTitle(): String = js.native
+
+
+  def getPath(): js.UndefOr[String] = js.native
+  def getURI(): js.UndefOr[String] = js.native
+  def insertNewline(): Unit = js.native
+  def isModified(): Boolean = js.native
+  def isEmpty(): Boolean = js.native
+  // def getEncoding(): Encoding = js.native
+  def setEncoding(encoding: String): Unit = js.native
+  def getTabLength(): Long = js.native
+
+  // File Operations
+  def save(): Unit = js.native
+  def saveAs(filePath: String): Unit = js.native
+
+  // Reading Text
+  def getText(): String = js.native
+  def getLineCount(): Long = js.native
+}
+  
+@js.native
+trait ConfigChange extends js.Object {
+  val newValue: js.Any = js.native
+  val oldValue: js.UndefOr[js.Any] = js.native
```

Comment:
The object passed to the callback in `onDidChange` always has both fields. So why `js.UndefOr[js.Any]`?